initrd-progs/0initrd/sbin/mountpartition

```diff
@@ -33,12 +33,35 @@ MNT_O='noatime'
 
 case $MNT_FS in
   ntfs)
-    ntfs-3g $MNT_DEV $MNT_DIR -o umask=0,no_def_opts,noatime,rw,silent 2>/dev/null #default is rw. 130211 add silent.
-    case $? in
-      0)  RET=0 ;;
-      14) echo "\\033[1;31m_WINDOWS_HIBERNATED_\\033[0;39m" >/dev/console ;; #31=red
-      *)  ntfs-3g $MNT_DEV $MNT_DIR -o umask=0,no_def_opts,noatime,rw,force,silent 2>/dev/null ;; #130211 add silent.
-    esac
+    
+    #Check if the kernel has builtin ntfs3 support
+    if [ "$(grep -m 1 'ntfs3' /proc/filesystems)" != "" ]; then
+     USE_NTFS3="y"
+     KVER=$(uname -r)
+     KMAJOR=$(echo "$KVER" | cut -f 1 -d '-' | cut -f 1 -d '.')
+     KMINOR=$(echo "$KVER" | cut -f 1 -d '-' | cut -f 2 -d '.')
+    else
+     USE_NTFS3=""
+    fi
+    
+    if [ "$USE_NTFS3" != "" ]; then
+       if [ $KMAJOR -ge 6 ] && [ $KMINOR -ge 2 ]; then
+        #Mount ntfs partition but respect windows filesystem setup. Only available on linux kernel 6.2 onwards
+        mount -t ntfs3 -o umask=0,noatime,prealloc,iocharset=utf8,nocase,windows_names $MNT_DEV $MNT_DIR
+       else
+        mount -t ntfs3 -o umask=0,noatime,prealloc,iocharset=utf8 $MNT_DEV $MNT_DIR
+       fi
+    else
```

Comment:
@rizalmart Why not simplify and just do this?

```
mount -t ntfs3 -o umask=0,noatime,prealloc,iocharset=utf8,nocase,windows_names $MNT_DEV $MNT_DIR || \
mount -t ntfs3 -o umask=0,noatime,prealloc,iocharset=utf8 $MNT_DEV $MNT_DIR || \
ntfs-3g $MNT_DEV $MNT_DIR -o umask=0,no_def_opts,noatime,rw,silent 2>/dev/null #default is rw. 130211 add silent.
``` 

(Without checking the kernel version or if it's a built-in driver - mount will fail and we'll fall back to the next option)

Reply:
Thats great idea. It is now implemented